database/schema.sql

```diff
@@ -44,6 +44,13 @@ CREATE TABLE invalidations (
     PRIMARY KEY (block_number, log_index)
 );
 
+-- MinFee Measurements from the API
+CREATE TABLE min_fee_measurements (
+  token bytea NOT NULL,
+  expiration_timestamp timestamptz NOT NULL,
+  min_fee numeric(78,0) NOT NULL
+);
```

Comment:
One thing I realize now is that the current design creates a new row every time queries the fee, which isn't great.
Maybe we could avoid this by checking whether the new fee is the same as the previous fee and in that case extending expiration_timestamp instead.
Or setting the expiration timestamp in our response to the same that is already there up to some limit. So the total time would be reduced for the later request. If our normal fee is supposed to last 5 minutes then we could say we will reuse the for 1 minute so users would see expiration time's between 4 and 5 minutes. In this case we could do this only if the fee is the same or even if the fee has changed because it probably won't change drastically.
Would probably be good to talk about this before merging with the new schema, in case the discussion would make it change.

Reply:
I'd suggest the following strategy:

The timestamp we report to the frontend is always 1 minute from now. Internally, we persist new estimates with a validity of 2 minutes. When querying a min fee we check if we have an estimate in the DB that is still valid 1 minute from now, if true return it. Otherwise create and save a new estimate. This way we should only have one record per minute.

This logic can live entirely in the minFeeCalculator logic.

Reply:
Sounds good.